opl/opl_sdl.c

```diff
@@ -137,7 +138,23 @@ int OPL_FillBuffer(byte *buffer, int buffer_samples)
         }
 
         // Add emulator output to buffer.
-        OPL3_GenerateStream(&opl_chip, (Bit16s *)(buffer + filled * 4), nsamples);
+        Bit16s *cursor = (Bit16s *)(buffer + filled * 4);
+        for (int s = 0; s < nsamples; ++s)
+        {
+            Bit32s mix[2] = {0, 0};
+            for (int c = 0; c < num_opl_chips; ++c)
+            {
+                Bit16s sample[2];
+                OPL3_GenerateResampled(&opl_chips[c], sample);
+                mix[0] += sample[0];
+                mix[1] += sample[1];
+            }
+
+            cursor[0] = mix[0] < -32768 ? -32768 : mix[0] > 32767 ? 32767 : mix[0];
+            cursor[1] = mix[1] < -32768 ? -32768 : mix[1] > 32767 ? 32767 : mix[1];
```

Comment:
For these range checks, consider using the `BETWEEN` macro: `BETWEEN(-32768, 32767, mix[n])`

Reply:
Ah, I was wondering if there was a CLAMP-style macro. I'll add this.